frontend/src/components/PixelInspector.tsx

```
@@ -66,6 +84,46 @@ function lookupValue(
+      Number.isNaN(pxFloat) ||
+      Number.isNaN(pyFloat)
+    ) {
+      return null;
+    }
+    if (pxFloat < 0 || pxFloat >= width || pyFloat < 0 || pyFloat >= height) {
+      return null;
+    }
+    const px = Math.min(width - 1, Math.max(0, Math.floor(pxFloat)));
+    const py = Math.min(height - 1, Math.max(0, Math.floor(pyFloat)));
+    const val = raw[py * width + px];
+    if (val !== val) return null;
```

Biome (2.4.13)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)